pkg/utils/dangerous_secret_names.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+)
+
+var DANGEROUS_SECRET_NAMES = []string{
+	"PROMPT_COMMAND",
+	"LD_PRELOAD",
+	"LD_LIBRARY_PATH",
+	"WINDIR",
+	"USERPROFILE",
+	"DYLD_INSERT_LIBRARIES",
+}
+
+type DangerousSecretNameError struct {
+	dangerousNames []string
+	currentConfig  string
+}
+
+func (e *DangerousSecretNameError) Error() string {
+	return fmt.Sprintf("the following dangerous secret names have been found in %s config:\n%s", e.currentConfig, strings.Join(e.dangerousNames, "\n"))
+}
+
+func CheckForDangerousSecretNames(secrets map[string]string) *DangerousSecretNameError {
```

Comment:
Does this intentionally return a pointer?

Reply:
If I don't return it as a pointer I can't return nil which. Since `DangerousSecretNameError` implements the `Error` interface and the idiomatic way to check errors in go is `if err != nil` I need to return a pointer.

Reply:
If you search the codebase for `type Error struct` you can see how we're handling this with our other custom errors.

Reply:
Taking a step back, I was able to refactor this so that I don't need a custom error. The `dangerousSecretNames` field on the custom error was solely for the unit test, but that was a crutch to make the tests pass. Instead I can just return an `fmt.Errorf(...)` and modified the tests to check that each value in the expected dangerous secret names was contained in the error string 😄